examples/cra-translation/README.md

```diff
@@ -0,0 +1,95 @@
+# @ttoss/i18n
+
+## 📚 About
+
+<strong> @ttoss/i18n</strong> is a easiest way to use translations in your React application.
```

Comment:
I wouldn't say that. `@ttoss/i18n` is the way to integrate all ttoss - modules with the applications. If the user doesn't use any module, then maybe it isn't worth using `@ttoss/i18n`.

Reply:
### I've been thinking about how we could do this, and I believe this example below is a viable possibility.

---

How could we do it in the @ttoss/auth project

```tsx
import { useTranslation, I18nProvider } from '@ttoss/i18n';

const defaultTranslations = {
  'pt-BR': {
    'You are logged': 'Você está logado',
    'Username field is required': 'O campo usuário é obrigatório',
    'Password field is required': 'O campo senha é obrigatório',
  },
};

type AuthProviderProps = {
  translations?: I18nProvider['translations'];
  initialLocale?: I18nProvider['initialLocale'];
};

const AuthProvider: React.FC<AuthProviderProps> = ({
  children,
  translations,
  initialLocale,
}) => {
  const [user, setUser] = React.useState(null);
  const isAuthenticated = !!user;

  const signOut = () => setUser(null);

  return (
    <I18nProvider
      initialLocale={initialLocale || 'pt-BR'}
      translations={{ ...defaultTranslations, ...translations }}
    >
      <AuthContext.Provider value={{ signOut, isAuthenticated, user }}>
        {children}
      </AuthContext.Provider>
    </I18nProvider>
  );
};

export default App;

export { useTranslation };
```

---

How could we use the auth module with these modifications

```tsx
import { AuthProvider } from '@ttoss/auth';
import App from './App';

const translations = {
  'en-US': {
    'You are logged': 'You are logged',
    'Username field is required': 'Username field is required',
    'Password field is required': 'Password field is required',
  },
};

const Application = () => {
  return (
    <AuthProvider initialLocale="en-US" translations={translations}>
      <App />
    </AuthProvider>
  );
};

export default Application;
```

I set up the example above, if it's a good possibility, I can make the changes in @ttoss/auth.

Reply:
I think it isn't good to pass `translations` to all modules. Suppose we had N models, then we'd to implement this way:

```
const Application = () => {
  return (
    <AuthProvider initialLocale="en-US" translations={translations}>
    <Module1 initialLocale="en-US" translations={translations}>
    <Module2 initialLocale="en-US" translations={translations}>
    ...
      <App />
    ...
    </Module2> 
    </Module1> 
    </AuthProvider> 
  );
};
```

In my opinion, it's better to use a provider:

```
const Application = () => {
  return (
    <I18nProvider initialLocale="en-US" translations={translations}>
    <AuthProvider>
    <Module1>
    <Module2>
    ...
      <App />
    ...
    </Module2> 
    </Module1> 
    </AuthProvider> 
    </I18nProvider>
  );
};
```

Reply:
@arantespp I'm thinking about how we could do this, but still out of ideas. Would you have any tips?